File: Aquarium/Engine/Sources/Engine/State.swift
public enum State: CustomStringConvertible, CustomDebugStringConvertible, Equatable {
    case none
    case air
    case water

    var isNone: Bool { self == .none }

    var isFluid: Bool { self == .water || self == .air }

    var next: Self {
        switch self {
        case .none: return .none
        case .water: return .air
        case .air: return .water
        }
    }

    public var description: String {
        switch self {
        case .none: return " "
        case .air: return "×"
        case .water: return "■"
        }
    }

    public var debugDescription: String { description }
}

File: rust/crates/grid/src/state.rs
use std::fmt;

#[derive(Clone, PartialEq, Eq)]
pub enum State {
    Water,
    Air,
    None,
}

impl State {
    pub fn as_char(&self) -> char {
        match self {
            State::Water => '■',
            State::Air => '×',
            State::None => ' ',
        }
    }

    pub fn is_none(&self) -> bool {
        self == &State::None
    }
}

impl fmt::Debug for State {
    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
        write!(f, "{}", self.as_char())
    }
}
